cpp/src/Slice/Scanner.l

```diff
@@ -64,546 +41,545 @@ using namespace Slice;
 namespace Slice
 {
 
-//
 // Definitions for the case-insensitive keyword-token map.
-//
-typedef std::map<std::string, int> StringTokenMap;
-static StringTokenMap keywordMap;
+typedef map<string, int> StringTokenMap;
+StringTokenMap keywordMap;
 
-void initScanner();
 int checkKeyword(string&);
-int checkIdentifier(string&);
+int checkIdentifier(const string&);
+
+}
+
+// Stores the scanner's current column position. Flex also automatically
+// generates 'yylineno', which stores the scanner's current line number.
+int yycolno = 0;
+// Stores a copy of the filename that the scanner is currently scanning.
+string yyfilename;
+
+namespace
+{
+
+void nextLine(int = 1);
+int scanPosition(const char*);
+void setLocation(TokenContext*);
+void startLocation(TokenContext*);
+void endLocation(TokenContext*);
+
+void initScanner();
+void preAction();
```

Comment:
Looking at the commit that most of this was pulled from (https://github.com/zeroc-ice/ice/commit/865ff422fb4c935726b859d79b73f2788e3add70#diff-3c507567a987cdf989038742a6964ad6d947511ef75ea5ab823136035205722dR67) on line 67.

Was there a reason for removing our custom `fatalError` hookin?
It's definitely not the most important, but in the very rare case there's an internal compiler error, it augments the default error message with some extra information that made debugging much easier for me in the past, before having to ask them to compile with tracing on.

Reply:
Visual Studio reports the function was never used